### ISSUE TITLE
JavaScript Data files are unreachable with absolute path `input`

### ISSUE THREAD

**Issue by brettinternet** [11ty/eleventy#1409]
### Describe the bug

When `input` receives an absolute path for its value, JavaScript data file values are imported as empty objects rather than exposing their exported values.

### To Reproduce

First, use an absolute path for `input` in the eleventy config.

```js
module.exports = {
  dir: {
    input: path.resolve("src"), // now JavaScript data files are imported as empty `{}`
  },
};
```

Then, add a JavaScript data file to `_data`, such as `project.js`.

Here is a simple reproduction of the issue: https://github.com/brettinternet/eleventy-bug-data-dir

### Expected Behavior

If this is expected behavior, perhaps we could improve the documentation on the `docs/config` page. I'd be happy to create a PR. It was clearly indicated in the documentation that the `data` directory must be relative to the `input` directory. However, there is no indication that `input` must be a relative path (although perhaps the default `input` value being `.` is some indication).

Otherwise, I would expect the JavaScript data files to work like other data file formats (I tried JSON and TOML) and make their contents available to the templating utilities. These other file formats worked as expected. See the demo repo for confirmation.

### Environment

- OS: `Linux`
- NPM: `6.14.7`
- Node: `13.9.0`
- Eleventy: `0.11.0`

I am unable to test on other OSs but my repo is available if others could test.

**Comment by brettinternet** [comment]
I've noticed that specifying a custom `includes` directory also benefited from a relative path (for hot reload to work with my pug partials).

**Comment by zachleat** [comment]
Confirmed this was fixed in 1.0
